### PR TITLE
Add Helix Sdk project and build telemetry tasks

### LINKED ISSUE #118: Produce tool for reporting builds to helix
Produce tool that can report build lifetime events to helix from a yaml build template.

### PR DESCRIPTION
**Pull request by alexperovich**
This change adds the Project for the Helix sdk and tasks to send
start and end build telemetry.

Fixes #118

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/HelixTask.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            if (!Directory.Exists(dir))
+            {
+                Directory.CreateDirectory(dir);
+            }
+
+            File.WriteAllText(ConfigFile, JsonConvert.SerializeObject(config, Constants.SerializerSettings));
+            return true;
+        }
+
+
```
Comment (jcagme) [inline src/Microsoft.DotNet.Helix/Sdk/HelixTask.cs]: Nit: remove blank line

File: src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.props [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.props]: Maybe add a comment like https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/sdk/Sdk.props#L3

File: src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.tasks [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.tasks]: This seems non-standard, why not move this into the props?

File: src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.targets [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <Import Project="$(MSBuildThisFileDirectory)Microsoft.DotNet.Helix.Sdk.tasks"/>
+
+  <Target Name="StartBuildTelemetry">
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.targets]: A comment defining usage and property value examples might be useful

File: src/Microsoft.DotNet.Helix/Sdk/StartBuildTelemetry.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        protected override async Task<bool> ExecuteCore()
+        {
+            var propertiesDict = new Dictionary<string, string>();
+            if (Properties != null)
+            {
+                foreach (string prop in Properties)
+                {
+                    string[] parts = prop.Split('=');
+                    if (parts.Length != 2)
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/StartBuildTelemetry.cs]: Is this preferrable over splitting on the first separator only? https://msdn.microsoft.com/en-us/library/c1bs0eda(v=vs.110).aspx

File: src/Microsoft.DotNet.Helix/Sdk/HelixTask.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        protected IHelixApi HelixApi { get; private set; }
+
+        private IHelixApi GetHelixApi()
+        {
+            if (string.IsNullOrEmpty(AccessToken))
+            {
+                return ApiFactory.GetAnonymous(BaseUri.AbsoluteUri);
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/HelixTask.cs]: Add a low importance log message for authenticated vs non-authenticated?

File: src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.tasks (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.tasks]: Sure

File: src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.targets (hunk 4 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.targets]: Added comments

**Comment by alexperovich** [comment]
@natemcmaster @weshaggard @mmitche do you guys have any comments? I would like input from stakeholders about this PR.

**Pull request merged**